microservices/case-service/src/main/java/org/entando/bundle/service/impl/CaseServiceImpl.java

```diff
@@ -141,24 +167,72 @@ private boolean deleteCaseResources(Case aCase) {
     List<Resource> resources = aCase != null && aCase.getMetadata() != null && aCase.getMetadata().getResources() != null ? aCase.getMetadata().getResources() : null;
 
     if (resources != null && !resources.isEmpty()) {
-        resources.forEach(r ->{
-          boolean res = fileService.deleteFile(r.getKey());
-
-          if (res) {
-            log.info("resource {} successfully removed from aCase {}", r.getKey(), aCase.getId());
-          } else {
-            log.warn("could not remove resource {} from aCase {}", r.getKey(), aCase.getId());
-          }
-          deleted[0] = deleted[0] && res;
-        });
+      resources.forEach(r ->{
+        boolean res = fileService.deleteFile(r.getKey());
+
+        if (res) {
+          log.info("resource {} successfully removed from aCase {}", r.getKey(), aCase.getId());
+        } else {
+          log.warn("could not remove resource {} from aCase {}", r.getKey(), aCase.getId());
+        }
+        deleted[0] = deleted[0] && res;
+      });
     }
     return deleted[0];
   }
 
-  public void stopProcess() {
+  /**
+   * Update the case with the real state of the underlying process
+   * @param aCase the case to be updated
+   * @return the updated Case object
+   */
+  protected Case syncWithProcessData(Case aCase) {
+    if (aCase != null && aCase.getState() != State.DELETED) {
+      String piid = aCase.getProcessInstanceId();
+
+      log.info("syncing data of case {} with process instance id {} ", aCase.getId(), aCase.getProcessInstanceId());
+      if (StringUtils.isNotBlank(piid)) {
+        if (bpmService.isProcessRunning(piid)) {
+          aCase.setState(State.RUNNING);
+        } else {
+          aCase.setState(State.COMPLETED);
+        }
+        // metadata
+        CaseMetadata metadata = aCase.getMetadata();
+        Map<String, Object> properties = bpmService.getProcessData(piid);
+        if (!properties.isEmpty()) {
+          log.debug("properties of process {} updated int case {}", aCase.getProcessInstanceId(), aCase.getId());
+          metadata.setProcessData(properties);
+        } else {
+          log.debug("no properties to attach to case {}", aCase.getId());
+        }
+      }
+    }
+    return aCase;
   }
 
-  public void resumeProcess() {
+  @Override
+  @Transactional
+  public void completeTaskState(final Long id, Map<String, Object> props) {
+    if (id != null) {
+      Optional<Case> optCase = getCase(id);
+      completeTaskState(optCase, props);
+    }
   }
 
+  @Override
+  public void completeTaskState(Optional<Case> optCase, Map<String, Object> props) {
+    if (optCase.isPresent()) {
+
+      if (props == null) {
+        props = new HashMap<>();
+      }
+      Case cur = optCase.get();
+      String instanceId = cur.getProcessInstanceId();
+      // update /add the current date time
+      props.put(VAR_UPDATED, LocalDateTime.now());
```

Comment:
VAR_UPDATED is a bit cryptic. Use PROCESS_INSTANCE_VARIABLES_LAST_UPDATE or smth similar

Reply:
Done